### PR TITLE
Setting up CI for ios

### PR DESCRIPTION
**Pull request by juanky201271**
There are a couple of old tests translated to swift, and working. CI now have IOS tests integrated as well.

### CODE REVIEW

File: .github/workflows/ios-build.yaml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        run: |
+          echo "zingolib_ref=$(echo ${GITHUB_REF} | sed 's/\//\\\//g')" >> $GITHUB_ENV
+          sudo sed -i "/^zingolib\|^zingoconfig/ s/branch = \"dev\"/rev = \"${{ env.zingolib_ref }}\"/" "rust/lib/Cargo.toml"
+          if cat rust/lib/Cargo.toml | grep "^zingolib\|^zingoconfig" | grep -q "branch"; then exit 1; fi
+          cd rust
+          sudo cargo update -p zingolib -p zingoconfig --aggressive
+          sudo rustup default nightly
+
+      - name: Install Cargo Lipo
+        run: |
```
Comment (Oscar-Pepper) [inline .github/workflows/ios-build.yaml:70]: for one liners this `|` isnt necessary

File: .github/workflows/ios-build.yaml [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+          sudo cargo install cargo-lipo
+
+      - name: Install rust Target
+        run: |
```
Comment (Oscar-Pepper) [inline .github/workflows/ios-build.yaml:74]: same as above

File: .github/workflows/ios-integration-test.yaml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        uses: actions/cache/restore@v4
+        with:
+          path: rust/lib/Generated
+          key: native-ios-uniffi-universal-${{ env.CACHE-KEY }}
+          fail-on-cache-miss: true
+
+      - name: Copy libs to ios folder
+        working-directory: rust/lib/Generated
+        run: |
+          cp ./*.* ../../../ios
```
Comment (Oscar-Pepper) [inline .github/workflows/ios-integration-test.yaml:51]: same as above

File: .github/workflows/ios-integration-test.yaml [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+      - name: Yarn Install
+        run: |
+          yarn
```
Comment (Oscar-Pepper) [inline .github/workflows/ios-integration-test.yaml:55]: same as above

File: .github/workflows/ios-integration-test.yaml [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+
+      - name: Pod Install
+        working-directory: ios
+        run: |
+          pod install
```
Comment (Oscar-Pepper) [inline .github/workflows/ios-integration-test.yaml:60]: same as above

**Review by Oscar-Pepper — APPROVED** [review]
AMAZING

File: .github/workflows/ios-build.yaml (hunk 1 above)
Comment (juanky201271) [reply .github/workflows/ios-build.yaml:70]: I will fix those in another PR, thanks.

**Pull request merged**